ddl/table.go

```diff
@@ -236,3 +214,65 @@ func (d *ddl) onTruncateTable(t *meta.Meta, job *model.Job) error {
 	job.Args = []interface{}{startKey}
 	return nil
 }
+
+func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) error {
+	var oldSchemaID int64
+	var tableName model.CIStr
+	if err := job.DecodeArgs(&oldSchemaID, &tableName); err != nil {
+		// Invalid arguments, cancel this job.
+		job.State = model.JobCancelled
+		return errors.Trace(err)
+	}
+
+	tblInfo, err := getTableInfo(t, job, oldSchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	newSchemaID := job.SchemaID
+	err = checkTableNotExists(t, job, newSchemaID, tblInfo.Name.L)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = t.DropTable(oldSchemaID, tblInfo.ID)
+	if err != nil {
+		job.State = model.JobCancelled
+		return errors.Trace(err)
+	}
+	tblInfo.Name = tableName
+	err = t.CreateTable(newSchemaID, tblInfo)
+	if err != nil {
+		job.State = model.JobCancelled
+		return errors.Trace(err)
+	}
+
+	ver, err := updateSchemaVersion(t, job)
+	if err != nil {
+		return errors.Trace(err)
```

Comment:
If it can pass tests, we can ensure that this function doesn't return the error.